tasks/watch.cr

```diff
@@ -41,18 +43,31 @@ module Sentry
 
       self.successful_compilations += 1
       if reload_browser?
-        reload_or_start_browser_sync
+        begin
+          reload_or_start_browser_sync
+        rescue Sentry::BrowserSyncPortInUseError
+          puts "Another process is currently using the BrowserSync port #{BROWSERSYNC_PORT}".colorize(:red)
+        end
       end
     end
 
     private def reload_or_start_browser_sync
       if successful_compilations == 1
+        check_browsersync_port_availability
         start_browsersync
       else
         reload_browsersync
       end
     end
 
+    private def check_browsersync_port_availability
+      if File.executable?(`which lsof`.chomp)
+        io = IO::Memory.new
+        Process.run("lsof -i :#{BROWSERSYNC_PORT}", output: io, error: STDERR, shell: true)
+        raise Sentry::BrowserSyncPortInUseError.new unless io.to_s.empty?
```

Comment:
Rather than raising an error and rescuing it, what do you think about renaming this method to `browsersync_port_is_available?` and return a Bool. Then create the processes or print the error. I think that's faster and a bit more straightforward than raising and rescuing an error

Reply:
Updated.